remove gist description from single file raw view (#2886)

Ma3oBblu opened the pull request:
this closes #2886

Ma3oBblu committed:
remove gist description from single file raw view (#2886)

mislav commented on pkg/cmd/gist/view/view.go:
```diff
@@ -107,6 +102,11 @@ func viewRun(opts *ViewOptions) error {
 	showFilenames := len(gist.Files) > 1
 
+	cs := opts.IO.ColorScheme()
+	if gist.Description != "" && showFilenames {
```
The new condition is going to make the rendering of gist description toggle on/off based on the number of files in a gist. This behavior might be confusing to users.

How about we do not render the description when a filename is specified, and always render otherwise?

Ma3oBblu replied:
change condition to check opts.Filename

mislav reviewed (changes requested):
This looks good! A couple of tweaks would still be needed.

In single-file mode, since we no longer need newlines to separate contents of different files, could we avoid printing `\n\n` below?
```go
out += fmt.Sprintf("%s\n\n", content)
```

Ma3oBblu committed:
change condition for single file, remove empty lines in single file mode

Ma3oBblu committed:
linter fixes

Ma3oBblu commented:
> This looks good! A couple of tweaks would still be needed.
> 
> In single-file mode, since we no longer need newlines to separate contents of different files, could we avoid printing `\n\n` below?
> 
> ```go
> out += fmt.Sprintf("%s\n\n", content)
> ```

fixed

mislav committed:
Improve `gist view` rendering

- Separate out logic to render a single file
- Render directly to stdout instead of to string slice
- Normalize whitespace between files; ensure no excessive trailing whitespace
- Add terminal pager support
- Sentence-case for flags

mislav reviewed (approved):
Fantastic! Thank you for the updates
